nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/nifi/pg/PGChangeAllVersions.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.nifi.toolkit.cli.api.CommandException;
+import org.apache.nifi.toolkit.cli.api.Context;
+import org.apache.nifi.toolkit.cli.impl.client.nifi.FlowClient;
+import org.apache.nifi.toolkit.cli.impl.client.nifi.NiFiClient;
+import org.apache.nifi.toolkit.cli.impl.client.nifi.NiFiClientException;
+import org.apache.nifi.toolkit.cli.impl.command.CommandOption;
+import org.apache.nifi.toolkit.cli.impl.command.nifi.AbstractNiFiCommand;
+import org.apache.nifi.toolkit.cli.impl.result.nifi.ProcessGroupsResult;
+import org.apache.nifi.toolkit.cli.impl.result.nifi.ProcessGroupsVersionChangeResult;
+import org.apache.nifi.toolkit.cli.impl.result.nifi.ChangeVersionResult;
+import org.apache.nifi.web.api.dto.ProcessGroupDTO;
+import org.apache.nifi.web.api.entity.VersionControlInformationEntity;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Command to change the version of a version controlled process group.
+ */
+public class PGChangeAllVersions extends AbstractNiFiCommand<ProcessGroupsVersionChangeResult> {
+
+    public PGChangeAllVersions() {
+        super("pg-change-all-versions", ProcessGroupsVersionChangeResult.class);
+    }
+
+    @Override
+    public String getDescription() {
+        return "Changes the version for all of the controlled process group instances for a given flow ID. "
+                + "This can be used to upgrade all the instances of a versioned flow to a new version, or "
+                + "revert to a previous version. If no version is specified, the latest version will be used. "
+                + "If no process group ID is provided, the root process group will be used to recursively "
+                + "search for all instances of the Flow ID. It is possible to force the recursive operation "
+                + "and not stop the operation in case the upgrade of a process group fails.";
+    }
+
+    @Override
+    protected void doInitialize(final Context context) {
+        addOption(CommandOption.FLOW_ID.createOption());
+        addOption(CommandOption.FLOW_VERSION.createOption());
+        addOption(CommandOption.PG_ID.createOption());
+        addOption(CommandOption.FORCE.createOption());
+    }
+
+    @Override
+    public ProcessGroupsVersionChangeResult doExecute(final NiFiClient client, final Properties properties)
+            throws NiFiClientException, IOException, MissingOptionException, CommandException {
+
+        final FlowClient flowClient = client.getFlowClient();
+        final String flowId = getRequiredArg(properties, CommandOption.FLOW_ID);
+
+        // get the optional id of the parent PG, otherwise fallback to the root group
+        String parentPgId = getArg(properties, CommandOption.PG_ID);
+        if (StringUtils.isBlank(parentPgId)) {
+            parentPgId = flowClient.getRootGroupId();
+        }
+
+        PGList doPGList = new PGList();
```

Comment:
We can have a few more `final` variables here, such as `doPGList`, `pgList`, and `doPGChangeVersion`.